huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_instances.go

```diff
@@ -0,0 +1,317 @@
+							Computed: true,
+						},
+						"user_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manegement_connect_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func isPubilcIPEnabled(val instances.Instance) bool {
+	return val.EnablePublicIP
+}
+
+func flattenKafkaInstanceList(client *golangsdk.ServiceClient, conf *config.Config, region string,
+	list []instances.Instance) ([]map[string]interface{}, []string, error) {
+	ids := make([]string, len(list))
+	result := make([]map[string]interface{}, len(list))
+
+	for i, val := range list {
+		partitionNum, err := strconv.ParseInt(val.PartitionNum, 10, 64)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// convert the ids of the availability zone into codes
+		availableZoneIDs := val.AvailableZones
+		availableZoneCodes, err := getAvailableZoneCodeByID(conf, region, availableZoneIDs)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		instance := map[string]interface{}{
+			"id":                         val.InstanceID,
+			"availability_zones":         availableZoneCodes,
+			"enterprise_project_id":      val.EnterpriseProjectID,
+			"product_id":                 val.ProductID,
+			"engine_version":             val.EngineVersion,
+			"storage_spec_code":          val.StorageSpecCode,
+			"storage_space":              val.TotalStorageSpace,
+			"type":                       val.Type,
+			"name":                       val.Name,
+			"description":                val.Description,
+			"vpc_id":                     val.VPCID,
+			"network_id":                 val.SubnetID,
+			"security_group_id":          val.SecurityGroupID,
+			"manager_user":               val.KafkaManagerUser,
+			"access_user":                val.AccessUser,
+			"maintain_begin":             val.MaintainBegin,
+			"maintain_end":               val.MaintainEnd,
+			"retention_policy":           val.RetentionPolicy,
+			"dumping":                    val.ConnectorEnalbe,
+			"enable_auto_topic":          val.EnableAutoTopic,
+			"partition_num":              partitionNum,
+			"ssl_enable":                 val.SslEnable,
+			"used_storage_space":         val.UsedStorageSpace,
+			"connect_address":            val.ConnectAddress,
+			"port":                       val.Port,
+			"status":                     val.Status,
+			"resource_spec_code":         val.ResourceSpecCode,
+			"user_id":                    val.UserID,
+			"user_name":                  val.UserName,
+			"manegement_connect_address": val.ManagementConnectAddress,
+			"tags":                       utils.TagsToMap(val.Tags),
+		}
+		if isPubilcIPEnabled(val) {
+			addrList := strings.Split(strings.TrimSpace(val.PublicConnectionAddress), ",")
+			log.Printf("[DEBUG] The address list is: %v", addrList)
```

Comment:
add  `PublicConnectionAddress` to the export attributes

Reply:
done.